kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java

```diff
@@ -293,16 +293,21 @@ protected void processFormLayout(FormInstance topLevelForm,
             List<String> scriptDataList) {
         FormLayout layout = form.getLayout();
         
+
         if (form.getModel().getClassName() != null && wrapJson) {
-            jsonTemplate                
-                .append("'")
-                .append(form.getModel().getName())
-                .append("' : ")
-                .append("{")
-                .append("'")
-                .append(form.getModel().getClassName())                
-                .append("' : ")
-                .append("{");
+            Optional<FormField> fieldForm = topLevelForm.getFields().stream().filter(e -> e.getNestedForm().equals(form.getId())).findAny();
+            jsonTemplate.append("'");
+            if (fieldForm.isPresent()) {
+                jsonTemplate.append(fieldForm.get().getBinding());
```

Comment:
this would be ideal to be already added to the FormModel class next to classname and name. Have you looked if this can be achieved?

Reply:
The problem is that FormModel is "clean" in the sense that the binding is introduced in the top level form and not here so you only have local information. Those comes from unmarshalling files.

Reply:
ok, fair enough